expression/builtin_vectorized.go

```diff
@@ -115,18 +115,24 @@ func (c *vecRowConverter) evalInt(row chunk.Row) (val int64, isNull bool, err er
 	if !c.builtinFunc.vectorized() {
 		return c.builtinFunc.evalInt(row)
 	}
-	var sel [1]int
-	sel[0] = row.Idx()
 	var buf *chunk.Column
 	if buf, err = c.builtinFunc.get(types.ETInt, 1); err != nil {
 		return
 	}
 
 	input := row.Chunk()
-	oriSel := input.Sel()
-	input.SetSel(sel[:])
-	err = c.builtinFunc.vecEvalInt(input, buf)
-	input.SetSel(oriSel)
+	if input == nil {
+		// it happens when the user ensure the leaf node is a Constant and then put an empty Row:
+		//   ConstantExpr.EvalInt(chunk.Row{})
+		err = c.builtinFunc.vecEvalInt(chunk.VirChunk, buf)
```

Comment:
We may call `err = c.builtinFunc.vecEvalInt(nil, buf)` here.
And check whether input == nil in Constant.VecEvalXXX

Reply:
Good suggestion.
It has been updated.